libfwupdplugin/fu-drm-device.c

```diff
@@ -206,9 +295,46 @@ fu_drm_device_probe(FuDevice *device, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_drm_device_setup(FuDevice *device, GError **error)
+{
+#ifdef HAVE_LIBDRM
+	FuDrmDevice *self = FU_DRM_DEVICE(device);
+	FuDrmDevicePrivate *priv = GET_PRIVATE(self);
+	gint fd;
+	g_autoptr(drmModeRes) res = NULL;
+
+	/* get crtcs */
+	fd = fu_io_channel_unix_get_fd(fu_udev_device_get_io_channel(FU_UDEV_DEVICE(device)));
+	res = drmModeGetResources(fd);
+	if (res == NULL) {
+		g_set_error_literal(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_NOT_SUPPORTED,
+				    "failed to get DRM resources");
+		return FALSE;
+	}
+	for (gint i = 0; i < res->count_crtcs; ++i) {
+		g_autoptr(drmModeCrtc) crtc = drmModeGetCrtc(fd, res->crtcs[i]);
+		if (crtc == NULL)
+			continue;
+		if (crtc->mode_valid && (crtc->mode.type & DRM_MODE_TYPE_PREFERRED) > 0) {
+			priv->crtc_x = crtc->x;
+			priv->crtc_y = crtc->y;
+			priv->crtc_width = crtc->width;
+			priv->crtc_height = crtc->height;
+			break;
+		}
+	}
+#endif
```

Comment:
to make this more right in more cases; what do you think about making two passes?  One that looks for eDP first, and then if no eDP found then anything with a preferred mode.

Reply:
I don't think eDP actually means anything special with regards to which crctc is "correct". I think the order is probably more correct.